doh-server/ietf.go

```diff
@@ -133,10 +134,19 @@ func (s *Server) parseRequestIETF(ctx context.Context, w http.ResponseWriter, r
 			if ipv4 := ednsClientAddress.To4(); ipv4 != nil {
 				ednsClientFamily = 1
 				ednsClientAddress = ipv4
-				ednsClientNetmask = 24
+				if s.conf.ECSFullSubnet {
+					ednsClientNetmask = 32
+				} else {
+					ednsClientNetmask = 24
+					ednsClientAddress = ednsClientAddress.Mask(net.CIDRMask(24, 32))
```

Comment:
Why put `ednsClientAddress.Mask` here but not in the IPv6 part below?

By the way, since I believe the DNS protocol only gives enough space to store the unmasked bytes, the `miekg/dns` library must be already removing the masked bytes before serializing. (Which means you can omit the `Mask` here safely.)

Reply:
done